### PR TITLE
[super-agent-deployment] First iteration of the super agent chart [NR-168648]

### PR DESCRIPTION
**Pull request by kang-makes**
#### Is this a new chart
Yes

#### What this PR does / why we need it:

This is the first iteration to start to work on the deployment of the super agent on Kubernetes.

#### Checklist
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[mychartname]`)

### CODE REVIEW

File: charts/super-agent-deployment/values.yaml [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+fedramp:
+  # -- (bool) Enables FedRAMP. Can be configured also with `global.fedramp.enabled`
+  # @default -- `false`
+  enabled:
+
+# -- (bool) Sets the debug logs to this integration or all integrations if it is set globally. Can be configured also with `global.verboseLog`
+# @default -- `false`
+verboseLog:
+
+# -- It holds the New Relic Prometheus configuration. Here you can easily set up Prometheus to get set metrics, discover
```
Comment (gsanchezgavier) [inline charts/super-agent-deployment/values.yaml:101]: some leftovers from base chart

File: charts/super-agent-deployment/values.yaml [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+fullnameOverride: ""
+
+# -- Name of the Kubernetes cluster monitored. Can be configured also with `global.cluster`.
+cluster: ""
+# -- This set this license key to use. Can be configured also with `global.licenseKey`
+licenseKey: ""
+# -- In case you don't want to have the license key in you values, this allows you to point to a user created secret to get the key from there. Can be configured also with `global.customSecretName`
+customSecretName: ""
+# -- In case you don't want to have the license key in you values, this allows you to point to which secret key is the license key located. Can be configured also with `global.customSecretLicenseKey`
+customSecretLicenseKey: ""
```
Comment (gsanchezgavier) [inline charts/super-agent-deployment/values.yaml:13]: Not sure if this is going to be implemented here (perhpas inside sub-agents configs), I think i would remove it to avoid noise or missleading config for now

File: charts/super-agent-deployment/values.yaml [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+# -- Annotations to be added to all pods created by the integration.
+podAnnotations: {}
+# -- Additional labels for chart pods. Can be configured also with `global.podLabels`
+podLabels: {}
+
+# -- Service that points to the super agent.
+# @default -- See `values.yaml`
+service:
+  type: ClusterIP
+  port: 80
```
Comment (gsanchezgavier) [inline charts/super-agent-deployment/values.yaml:51]: Same for this, not sure if we will us it , I would prefer to add it when is needed, unless you expect some issues doing it.

File: charts/super-agent-deployment/values.yaml [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+proxy: ""
+
+# -- (bool) Send the metrics to the staging backend. Requires a valid staging license key. Can be configured also with `global.nrStaging`
+# @default -- `false`
+nrStaging:
+
+fedramp:
+  # -- (bool) Enables FedRAMP. Can be configured also with `global.fedramp.enabled`
+  # @default -- `false`
+  enabled:
```
Comment (gsanchezgavier) [inline charts/super-agent-deployment/values.yaml:95]: I would add it in the future when implemented

**Review by gsanchezgavier — COMMENTED** [review]
thanks! it looks good to me, I would remove the things we are not expecting to implement in the near future (MVP) if that is not going to represent any issue, just to avoid us the cleaning work, or misleading configs

File: charts/super-agent-deployment/Chart.yaml [hunk 5]
```diff
@@ -0,0 +1,28 @@
+apiVersion: v2
+name: super-agent-deployment
+description: A Helm chart to install New Relic Super agent on Kubernetes
+
+type: application
+version: 0.0.0-beta
+appVersion: TODO-FILL-THIS-WITH-CI
```
Comment (paologallinaharbur) [inline charts/super-agent-deployment/Chart.yaml:7]: Can you add a comment (in code or in PR) explaining this TODO a bit more?

File: charts/super-agent-deployment/README.md [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+
+A Helm chart to install New Relic Super agent on Kubernetes
+
+# Helm installation
+
+You can install this chart using directly this Helm repository:
+
+```shell
+helm repo add newrelic https://newrelic.github.io/helm-charts
+helm upgrade --install newrelic/CHART-TEMPLATE -f your-custom-values.yaml
```
Comment (paologallinaharbur) [inline charts/super-agent-deployment/README.md:15]: CHART-TEMPLATE?

File: charts/super-agent-deployment/README.md [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+Options that can be defined globally include `affinity`, `nodeSelector`, `tolerations`, `proxy` and others. The full list can be found at
+[user's guide of the common library](https://github.com/newrelic/helm-charts/blob/master/library/common-library/README.md).
+
+## Chart particularities
+
+> **TODO:** Here is where you should add particularities for this chart like what does the chart do with the privileged and
+low data modes or any other quirk that it could have.
+
+### TODOs
+There are values that should be planned at some point of removed from the `values.yaml`. I leave them here documented
```
Comment (paologallinaharbur) [inline charts/super-agent-deployment/README.md:32]: > There are values that should be planned at some point of removed from the `values.yaml`

I would reword this sentence since it is a bit confusing

File: charts/super-agent-deployment/README.md [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+Other TODOs:
+ * Add probes for liveness and readiness.
+
+## Values
+
+| Key | Type | Default | Description |
+|-----|------|---------|-------------|
+| affinity | object | `{}` | Sets pod/node affinities. Can be configured also with `global.affinity` |
+| cluster | string | `""` | Name of the Kubernetes cluster monitored. Can be configured also with `global.cluster`. |
+| config | object | See `values.yaml` | It holds the New Relic Prometheus configuration. Here you can easily set up Prometheus to get set metrics, discover ponds and endpoints Kubernetes and send metrics to New Relic using remote-write. |
```
Comment (paologallinaharbur) [inline charts/super-agent-deployment/README.md:49]: Prometheus?

File: charts/super-agent-deployment/values.yaml (hunk 2 above)
Comment (paologallinaharbur) [reply charts/super-agent-deployment/values.yaml:13]: Actually the SuperAgent needs a license key, therefore it makes sense to have this, however I am not sure if it can be passed already as a envVar

File: charts/super-agent-deployment/templates/rbac.yaml [hunk 9]
```diff
@@ -0,0 +1,34 @@
+{{- if .Values.rbac.create }}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
```
[the remark this answers is not in the document]
Comment (kang-makes) [reply charts/super-agent-deployment/templates/rbac.yaml:4]: Fair enough. Solved in [638c463](https://github.com/newrelic/helm-charts/pull/1155/commits/638c4630bee6584651b64e1399a7e950e0686f4e)

File: charts/super-agent-deployment/values.yaml (hunk 1 above)
Comment (kang-makes) [reply charts/super-agent-deployment/values.yaml]: Oops. Solved on [6b86ff9](https://github.com/newrelic/helm-charts/pull/1155/commits/6b86ff9419c8a494ba7b7ddd8fd52a34798345f7)

File: charts/super-agent-deployment/templates/deployment-superagent.yaml [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+          #     path: /
+          #     port: http
+          # readinessProbe:
+          #   httpGet:
+          #     path: /
+          #     port: http
+
+          volumeMounts:
+            - name: super-agent-config
+              mountPath: /etc/newrelic-super-agent/config.yaml
```
Comment (gsanchezgavier) [inline charts/super-agent-deployment/templates/deployment-superagent.yaml:72]: https://github.com/newrelic/helm-charts/actions/runs/6563315348/job/17827275669?pr=1155#step:8:121
```suggestion
              mountPath: /etc/newrelic-super-agent/
```

File: charts/super-agent-deployment/README.md [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+| priorityClassName | string | `""` | Sets pod's priorityClassName. Can be configured also with `global.priorityClassName` |
+| proxy | string | `""` | Configures the integration to send all HTTP/HTTPS request through the proxy in that URL. The URL should have a standard format like `https://user:password@hostname:port`. Can be configured also with `global.proxy` |
+| rbac.create | bool | `true` | Whether the chart should automatically create the RBAC objects required to run. |
+| resources | object | `{}` | Resource limits to be added to all pods created by the integration. |
+| serviceAccount | object | See `values.yaml` | Settings controlling ServiceAccount creation. |
+| serviceAccount.create | bool | `true` | Whether the chart should automatically create the ServiceAccount objects required to run. |
+| tolerations | list | `[]` | Sets pod's tolerations to node taints. Can be configured also with `global.tolerations` |
+| verboseLog | bool | `false` | Sets the debug logs to this integration or all integrations if it is set globally. Can be configured also with `global.verboseLog` |
```
Comment (paologallinaharbur) [inline charts/super-agent-deployment/README.md:77]: does this do something already?

File: charts/super-agent-deployment/README.md [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+| fedramp.enabled | bool | `false` | Enables FedRAMP. Can be configured also with `global.fedramp.enabled` |
+| fullnameOverride | string | `""` | Override the full name of the release |
+| hostNetwork | bool | `false` | Sets pod's hostNetwork. Can be configured also with `global.hostNetwork` |
+| image | object | See `values.yaml` | Image for the New Relic Super Agent |
+| image.pullSecrets | list | `[]` | The secrets that are needed to pull images from a custom registry. |
+| labels | object | `{}` | Additional labels for chart objects. Can be configured also with `global.labels` |
+| licenseKey | string | `""` | This set this license key to use. Can be configured also with `global.licenseKey` |
+| nameOverride | string | `""` | Override the name of the chart |
+| nodeSelector | object | `{}` | Sets pod's node selector. Can be configured also with `global.nodeSelector` |
+| nrStaging | bool | `false` | Send the metrics to the staging backend. Requires a valid staging license key. Can be configured also with `global.nrStaging` |
```
Comment (paologallinaharbur) [inline charts/super-agent-deployment/README.md:66]: The superAgent does not send metrics

File: charts/super-agent-deployment/README.md [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+| containerSecurityContext | object | `{}` | Sets security context (at container level). Can be configured also with `global.containerSecurityContext` |
+| customAttributes | object | `{}` | Adds extra attributes to the cluster and all the metrics emitted to the backend. Can be configured also with `global.customAttributes` |
+| customSecretLicenseKey | string | `""` | In case you don't want to have the license key in you values, this allows you to point to which secret key is the license key located. Can be configured also with `global.customSecretLicenseKey` |
+| customSecretName | string | `""` | In case you don't want to have the license key in you values, this allows you to point to a user created secret to get the key from there. Can be configured also with `global.customSecretName` |
+| dnsConfig | object | `{}` | Sets pod's dnsConfig. Can be configured also with `global.dnsConfig` |
+| extraVolumeMounts | list | `[]` | Defines where to mount volumes specified with `extraVolumes` |
+| extraVolumes | list | `[]` | Volumes to mount in the containers |
+| fedramp.enabled | bool | `false` | Enables FedRAMP. Can be configured also with `global.fedramp.enabled` |
```
Comment (paologallinaharbur) [inline charts/super-agent-deployment/README.md:57]: I would not include in the docs features not implemented yet

File: charts/super-agent-deployment/values.yaml [hunk 14]
```diff
@@ -0,0 +1,103 @@
+# -- Override the name of the chart
+nameOverride: ""
+# -- Override the full name of the release
+fullnameOverride: ""
+
+# -- Name of the Kubernetes cluster monitored. Can be configured also with `global.cluster`.
+cluster: ""
```
Comment (paologallinaharbur) [inline charts/super-agent-deployment/values.yaml:7]: Is this used?

File: charts/super-agent-deployment/values.yaml [hunk 15]
```diff
@@ -0,0 +1,103 @@
...
+
+# -- Sets pod/node affinities. Can be configured also with `global.affinity`
+affinity: {}
+# -- Sets pod's node selector. Can be configured also with `global.nodeSelector`
+nodeSelector: {}
+# -- Sets pod's tolerations to node taints. Can be configured also with `global.tolerations`
+tolerations: []
+
+# -- Adds extra attributes to the cluster and all the metrics emitted to the backend. Can be configured also with `global.customAttributes`
+customAttributes: {}
```
Comment (paologallinaharbur) [inline charts/super-agent-deployment/values.yaml:77]: I would add this in the future

File: charts/super-agent-deployment/templates/deployment-superagent.yaml [hunk 16]
```diff
@@ -0,0 +1,92 @@
...
+
+          # TODO: Add probes
+          # livenessProbe:
+          #   httpGet:
+          #     path: /
+          #     port: http
+          # readinessProbe:
+          #   httpGet:
+          #     path: /
+          #     port: http
```
Comment (paologallinaharbur) [inline charts/super-agent-deployment/templates/deployment-superagent.yaml:68]: I would remove commented out code

File: charts/super-agent-deployment/values.yaml (hunk 2 above)
Comment (paologallinaharbur) [reply charts/super-agent-deployment/values.yaml:13]: Thinking about this again, are we adding the logic to leverage this?

If the super-agent supports it we need something like this https://github.com/newrelic/nri-kubernetes/blob/0c8d886317af6bcfeac7536565e654378da0b652/charts/newrelic-infrastructure/templates/kubelet/daemonset.yaml#L116

File: charts/super-agent-deployment/values.yaml [hunk 17]
```diff
@@ -0,0 +1,103 @@
...
+  tag: ""  # Defaults to chart's appVersion
+  imagePullPolicy: IfNotPresent
+  # -- The secrets that are needed to pull images from a custom registry.
+  pullSecrets: []
+  # - name: regsecret
+
+# -- Volumes to mount in the containers
+extraVolumes: []
+# -- Defines where to mount volumes specified with `extraVolumes`
+extraVolumeMounts: []
```
Comment (paologallinaharbur) [inline charts/super-agent-deployment/values.yaml:29]: are these used?

File: charts/super-agent-deployment/values.yaml (hunk 2 above)
Comment (paologallinaharbur) [reply charts/super-agent-deployment/values.yaml:13]: and the same for licenseKey and customSecretName

**Review by paologallinaharbur — COMMENTED** [review]
I agree with @gsanchezgavier I would remove from the values what is not available or implemented yet.

I would add the easy ones (extra volumes, ..) and the important ones (licenseKey secret, ...) and delete (or mark with TODO) the rest for now

File: charts/super-agent-deployment/templates/_helpers.tpl [hunk 18]
```diff
@@ -0,0 +1,23 @@
+{{- /*
+`newrelic-super-agent.config` builds the configuration from config on the values and add more config options like
+cluster name, licenses, and custom attributes
+*/ -}}
+{{- define "newrelic-super-agent.config" -}}
+{{- /*
+TODO:
+    * `licenseKey`: comes from the common library but it would not be needed as it is an ingestion API Key, not a REST one.
+    * `rbac`: the is a placeholder for RBAC that simply list pods. this has to be narrowed to the use case of this agent.
```
Comment (paologallinaharbur) [inline charts/super-agent-deployment/templates/_helpers.tpl:9]: this was implemented already, right?

File: charts/super-agent-deployment/values.yaml (hunk 4 above)
Comment (kang-makes) [reply charts/super-agent-deployment/values.yaml]: I would prefer, as a first iteration, to have marked everything that comes from the `common-library` as TODO to not forget that we are supporting that at all New Relic's charts. We can do a `grep TODO -R` at any point and see what is missing.

I added TODOs on [#1b027c0](https://github.com/newrelic/helm-charts/pull/1155/commits/1b027c0848604a02513c49424606eacb7d4e03ce)

Let me know what you think, I have no strong opinion.

File: charts/super-agent-deployment/Chart.yaml (hunk 5 above)
Comment (kang-makes) [reply charts/super-agent-deployment/Chart.yaml]: Done on [#6b31e5f](https://github.com/newrelic/helm-charts/pull/1155/commits/6b31e5fa129a97e2e585cf0117981ad3df4a4233)

File: charts/super-agent-deployment/values.yaml (hunk 4 above)
Comment (paologallinaharbur) [reply charts/super-agent-deployment/values.yaml]: I am fine with the TODOs 😄 Before releasing we can decide to address them or remove them

I would just remove what we know that we will never support 

@gsanchezgavier we could add it to the list of leftovers

File: charts/super-agent-deployment/README.md (hunk 6 above)
Comment (kang-makes) [reply charts/super-agent-deployment/README.md:15]: I expected that people would complain about the chart names. I was wrong so I changed the URL to the definitive one on [#d0c2eb5](https://github.com/newrelic/helm-charts/pull/1155/commits/d0c2eb57da5a26394ec0e5e59e173ac18425c5fb)

File: charts/super-agent-deployment/README.md (hunk 7 above)
Comment (kang-makes) [reply charts/super-agent-deployment/README.md]: As I documented the TODOs [here](https://github.com/newrelic/helm-charts/pull/1155#discussion_r1365573200) y removed the section from the README on [#0a14139](https://github.com/newrelic/helm-charts/pull/1155/commits/0a141392c9c249476c8ca69630b65908defff74a)

File: charts/super-agent-deployment/README.md (hunk 11 above)
Comment (kang-makes) [reply charts/super-agent-deployment/README.md]: No. I marked as TODO here: https://github.com/newrelic/helm-charts/pull/1155#discussion_r1365573200

File: charts/super-agent-deployment/README.md (hunk 12 above)
Comment (kang-makes) [reply charts/super-agent-deployment/README.md]: Then, why would we need a `licenseKey`? Remember that the `licenseKey` is not the REST API Key.

File: charts/super-agent-deployment/README.md (hunk 13 above)
Comment (kang-makes) [reply charts/super-agent-deployment/README.md]: It was marked as TODO here: https://github.com/newrelic/helm-charts/pull/1155#discussion_r1365584627

File: charts/super-agent-deployment/templates/_helpers.tpl [hunk 19]
```diff
@@ -0,0 +1,23 @@
...
+    * `verboseLog`: legacy toggle to enable verbosity.
+
+For this iteration, the chart has no way to template tehe license. You might reuse the common-library secret creation helpers:
+{{- include "newrelic.common.license.secret" . -}}
+
+*/ -}}
+
+{{ if .Values.config }}
+    {{- .Values.config | toYaml -}}
+{{- end -}}
```
[the remark this answers is not in the document]
Comment (kang-makes) [reply charts/super-agent-deployment/templates/_helpers.tpl:21]: We have still to be implemented `verboseLog`, `licenseKey` and many things from the common library so that helper is going to be needed.

I am still not sure how but it is going to happen for sure.

File: charts/super-agent-deployment/values.yaml (hunk 14 above)
Comment (kang-makes) [reply charts/super-agent-deployment/values.yaml:7]: Marked as TODO [here](https://github.com/newrelic/helm-charts/pull/1155#discussion_r1365584627).

File: charts/super-agent-deployment/values.yaml [hunk 20]
```diff
@@ -0,0 +1,103 @@
...
+# -- (bool) Sets the debug logs to this integration or all integrations if it is set globally. Can be configured also with `global.verboseLog`
+# @default -- `false`
+verboseLog:
+
+# -- Here you can set New Relic' Super Agent configuration.
+# @default -- See `values.yaml` for examples
+config: {}
+#  opamp:
+#    headers:
+#      some-key: some-value
```
[the remark this answers is not in the document]
Comment (kang-makes) [reply charts/super-agent-deployment/values.yaml:100]: I was still waiting to have time to investigate the comment you left: https://github.com/newrelic/helm-charts/pull/1155#discussion_r1363883747

I still have to raise this to CAOS.

File: charts/super-agent-deployment/values.yaml (hunk 17 above)
Comment (kang-makes) [reply charts/super-agent-deployment/values.yaml:29]: They are.

File: charts/super-agent-deployment/templates/deployment-superagent.yaml (hunk 10 above)
Comment (kang-makes) [reply charts/super-agent-deployment/templates/deployment-superagent.yaml]: Good catch. Solved here [#403ce07](https://github.com/newrelic/helm-charts/pull/1155/commits/403ce07302dd2d6189d019db329d7f06464857d6) and set as read-only.

File: charts/super-agent-deployment/README.md (hunk 12 above)
Comment (paologallinaharbur) [reply charts/super-agent-deployment/README.md]: RIght now the opamp connection is based on the licenseKey

File: charts/super-agent-deployment/templates/deployment-superagent.yaml (hunk 16 above)
Comment (paologallinaharbur) [reply charts/super-agent-deployment/templates/deployment-superagent.yaml]: I am fine with keeping it, I would just remove the example then

File: charts/super-agent-deployment/values.yaml (hunk 17 above)
Comment (paologallinaharbur) [reply charts/super-agent-deployment/values.yaml:29]: ahh ok I guess I missed them

File: charts/super-agent-deployment/templates/_helpers.tpl (hunk 19 above)
[the remark this answers is not in the document]
Comment (paologallinaharbur) [reply charts/super-agent-deployment/templates/_helpers.tpl:21]: ahh ok it makes sense! you are right, we should keep it!

File: charts/super-agent-deployment/templates/rbac.yaml [hunk 21]
```diff
@@ -0,0 +1,42 @@
...
+    {{- include "newrelic.common.labels" . | nindent 4 }}
+  name: {{ include "newrelic.common.naming.fullname" . }}
+  namespace: {{ .Release.Namespace }}
+rules:
+  - apiGroups:
+      - notification.toolkit.fluxcd.io
+      - source.toolkit.fluxcd.io
+      - helm.toolkit.fluxcd.io
+      - image.toolkit.fluxcd.io
+      - kustomize.toolkit.fluxcd.io
```
Comment (paologallinaharbur) [inline charts/super-agent-deployment/templates/rbac.yaml:16]: I know it is early to speak about this, but we are not expecting to create notification, image or customize CR

File: charts/super-agent-deployment/tests/__snapshot__/.gitignore
Comment (paologallinaharbur) [inline charts/super-agent-deployment/tests/__snapshot__/.gitignore:1]: is this a leftover?

File: charts/super-agent-deployment/templates/_helpers.tpl (hunk 18 above)
Comment (kang-makes) [reply charts/super-agent-deployment/templates/_helpers.tpl]: I removed that comment as every TODO is not commented on the `values.yaml` of the deployment template.

Removed on [#cef489e](https://github.com/newrelic/helm-charts/pull/1155/commits/cef489ec5fdd577cc4fc46010ee5d57b1e368705)

File: charts/super-agent-deployment/ci/test-values.yaml [hunk 22]
```diff
@@ -0,0 +1,3 @@
+global:
+  licenseKey: 1234567890abcdef1234567890abcdef12345678
+  cluster: test-cluster
```
[the remark this answers is not in the document]
Comment (kang-makes) [reply charts/super-agent-deployment/ci/test-values.yaml]: Solved on [#3c3578f](https://github.com/newrelic/helm-charts/pull/1155/commits/3c3578f9820de214bc5aeb9f12c9bc9de34163b9)

File: charts/super-agent-deployment/templates/deployment-superagent.yaml (hunk 16 above)
Comment (kang-makes) [reply charts/super-agent-deployment/templates/deployment-superagent.yaml]: Removed on [#3c097cf](https://github.com/newrelic/helm-charts/pull/1155/commits/3c097cf00104f992cfa83cf948302cd184772c81)

File: charts/super-agent-deployment/templates/rbac.yaml (hunk 21 above)
Comment (kang-makes) [reply charts/super-agent-deployment/templates/rbac.yaml]: Fair enough/ Removed on [#51cefaa](https://github.com/newrelic/helm-charts/pull/1155/commits/51cefaa5326aac44f1e1754647f953d3aed05a87)

File: charts/super-agent-deployment/tests/__snapshot__/.gitignore
Comment (kang-makes) [reply charts/super-agent-deployment/tests/__snapshot__/.gitignore:1]: It is not. For a glance, `helm unittest` creates a snapshot of the output of helm on that file that the developer can save with `--update-snapshot`.

The use-case is to test the whole set of documents instead of simply testing a field of it.

File: charts/super-agent-deployment/values.yaml (hunk 4 above)
Comment (kang-makes) [reply charts/super-agent-deployment/values.yaml]: Already marked as TODO: https://github.com/newrelic/helm-charts/pull/1155/files#diff-93b1805c025a55ca2f62fa3f3feaf426afb9b715fa6e8e2c63ab07c76a6da5afR87

I can mark this as solved :D

File: charts/super-agent-deployment/templates/deployment-superagent.yaml [hunk 23]
```diff
@@ -0,0 +1,88 @@
...
+
+      containers:
+        - name: {{ .Chart.Name }}
+          {{- with include "newrelic.common.securityContext.container" . }}
+          securityContext:
+            {{- . | nindent 12 }}
+          {{- end }}
+          image: {{ include "newrelic.common.images.image" ( dict "imageRoot" .Values.image "context" .) }}
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+
```
Comment (gsanchezgavier) [inline charts/super-agent-deployment/templates/deployment-superagent.yaml:54]: the running mode has already been merged
```
 args:
  - --running-mode=Kubernetes
```

File: charts/super-agent-deployment/Chart.yaml [hunk 24]
```diff
@@ -0,0 +1,28 @@
+apiVersion: v2
+name: super-agent-deployment
+description: A Helm chart to install New Relic Super agent on Kubernetes
+
+type: application
+version: 0.0.0-beta
+appVersion: nightly  # Change this with a proper version of the image.
```
Comment (paologallinaharbur) [inline charts/super-agent-deployment/Chart.yaml:7]: can you add TODO? just to be consistent?

File: charts/super-agent-deployment/templates/deployment-superagent.yaml (hunk 23 above)
Comment (kang-makes) [reply charts/super-agent-deployment/templates/deployment-superagent.yaml:56]: Done on [#8251ec6](https://github.com/newrelic/helm-charts/pull/1155/commits/8251ec6767fd6a5b79b7864f81a0ca8b2af6f3d1)

File: charts/super-agent-deployment/Chart.yaml (hunk 24 above)
Comment (kang-makes) [reply charts/super-agent-deployment/Chart.yaml:7]: 🤦

File: charts/super-agent-deployment/templates/configmap-superagent-config.yaml [hunk 25]
```diff
@@ -0,0 +1,15 @@
+{{- $name := include "newrelic.common.naming.truncateToDNSWithSuffix" ( dict "name" (include "newrelic.common.naming.fullname" .) "suffix" "config" ) -}}
+{{- $namespace := .Release.Namespace -}}
+{{- with ( include "newrelic-super-agent.config" . ) -}}
```
Comment (paologallinaharbur) [inline charts/super-agent-deployment/templates/configmap-superagent-config.yaml:3]: using `if` instead of `with` could we save two variables?

File: charts/super-agent-deployment/values.yaml [hunk 26]
```diff
@@ -0,0 +1,103 @@
...
+  # @default -- `false`
+  enabled:
+
+# -- (bool) TODO: Sets the debug logs to this integration or all integrations if it is set globally. Can be configured also with `global.verboseLog`
+# @default -- `false`
+verboseLog:
+
+# -- Here you can set New Relic' Super Agent configuration.
+# @default -- See `values.yaml` for examples
+config: {}
```
Comment (paologallinaharbur) [inline charts/super-agent-deployment/values.yaml:97]: Notice that apart from this config we will have one config for each sub-agent. Since it is a bit too early I think we can skip it and add it in the future whenever we do the config task with Mark.

@kang-makes @gsanchezgavier what do you think?

File: charts/super-agent-deployment/templates/rbac.yaml [hunk 27]
```diff
@@ -0,0 +1,40 @@
...
+metadata:
+  labels:
+    {{- include "newrelic.common.labels" . | nindent 4 }}
+  name: {{ include "newrelic.common.naming.fullname" . }}
+  namespace: {{ .Release.Namespace }}
+rules:
+  - apiGroups:
+      - source.toolkit.fluxcd.io
+      - helm.toolkit.fluxcd.io
+      - kustomize.toolkit.fluxcd.io
```
Comment (paologallinaharbur) [inline charts/super-agent-deployment/templates/rbac.yaml:14]: Kustomize is disabled for now, we could remove this as well I think, unless you think it could be useful for some workarounds

**Review by paologallinaharbur — APPROVED** [review]
Apart from some NITs it is good to go from my point of view!

File: charts/super-agent-deployment/templates/configmap-superagent-config.yaml (hunk 25 above)
Comment (kang-makes) [reply charts/super-agent-deployment/templates/configmap-superagent-config.yaml]: I already did it on [#7ec3d4b](https://github.com/newrelic/helm-charts/pull/1155/commits/7ec3d4ba9073b07f1ca579fc3547ebcb10695eef).

File: charts/super-agent-deployment/templates/rbac.yaml (hunk 27 above)
Comment (kang-makes) [reply charts/super-agent-deployment/templates/rbac.yaml]: Done in [#2e7ea0a](https://github.com/newrelic/helm-charts/pull/1155/commits/2e7ea0a72b79ef91f8fdc94f8231798b34553883)

File: charts/super-agent-deployment/templates/deployment-superagent.yaml [hunk 28]
```diff
@@ -0,0 +1,90 @@
...
+      hostNetwork: {{ include "newrelic.common.hostNetwork.value" . }}
+      {{- if include "newrelic.common.hostNetwork" . }}
+      dnsPolicy: ClusterFirstWithHostNet
+      {{- end }}
+
+      serviceAccountName: {{ include "newrelic.common.serviceAccount.name" . }}
+
+      containers:
+        - name: {{ .Chart.Name }}
```
[the remark this answers is not in the document]
Comment (kang-makes) [reply charts/super-agent-deployment/templates/deployment-superagent.yaml:47]: Done in [#2e7ea0a](https://github.com/newrelic/helm-charts/pull/1155/commits/2e7ea0a72b79ef91f8fdc94f8231798b34553883)

File: charts/super-agent-deployment/Chart.yaml (hunk 24 above)
Comment (kang-makes) [reply charts/super-agent-deployment/Chart.yaml]: Done in [#2e7ea0a](https://github.com/newrelic/helm-charts/pull/1155/commits/2e7ea0a72b79ef91f8fdc94f8231798b34553883)

File: charts/super-agent-deployment/values.yaml (hunk 20 above)
[the remark this answers is not in the document]
Comment (kang-makes) [reply charts/super-agent-deployment/values.yaml]: This was completely redone in [#2e7ea0a](https://github.com/newrelic/helm-charts/pull/1155/commits/2e7ea0a72b79ef91f8fdc94f8231798b34553883)

File: charts/super-agent-deployment/README.md (hunk 12 above)
Comment (kang-makes) [reply charts/super-agent-deployment/README.md:61]: And I added support for it in [#2e7ea0a](https://github.com/newrelic/helm-charts/pull/1155/commits/2e7ea0a72b79ef91f8fdc94f8231798b34553883)

**Comment by kang-makes** [comment]
I am not going to be able to make tests pass on this chart without modifying the CI to add a valid licenseKey or a mock OpAmp server to the cluster.

If I remove the `ci/*` values `chart-tester` still uses the default ones that fail because it is not able to template without a license.

If I add testing values, they have invalid licenseKey hence failing with a 403.

This chart has to be merged like this for now.

File: charts/super-agent-deployment/templates/_helpers.tpl [hunk 29]
```diff
@@ -0,0 +1,72 @@
...
+
+{{- /*
+This function simply templates the default configuration for the agent.
+*/ -}}
+{{- define "newrelic-super-agent.config.defaultConfig" -}}
+opamp:
+  endpoint: COMPLETE-ME
+  headers:
+    api-key: COMPLETE-ME
+{{- end -}}
```
Comment (paologallinaharbur) [inline charts/super-agent-deployment/templates/_helpers.tpl:25]: why are we not putting this as a default of the values.yaml for `newrelic-super-agent.config.content` for now? 

If I am not mistaken we are going always to override that 🤔

File: charts/super-agent-deployment/templates/_helpers.tpl [hunk 30]
```diff
@@ -0,0 +1,72 @@
...
+
+If you need a list of TODOs, just `grep TODO` on the `values.yaml` and look for things that are yet to be implemented.
+*/ -}}
+{{- $config := fromYaml (include "newrelic-super-agent.config.defaultConfig" .) -}}
+{{- if .Values.config.content -}}
+  {{- $_ := deepCopy .Values.config.content | mustMergeOverwrite $config -}}
+{{- end -}}
+
+{{- if include "newrelic.common.fedramp.enabled" . -}}
+    {{- fail "FedRAMP is not supported yet" -}}{{- /* TODO: Add FedRamp support */ -}}
```
Comment (paologallinaharbur) [inline charts/super-agent-deployment/templates/_helpers.tpl:59]: fedramp is not marked as TODO in readme, if it is not supported why checking for the bool?

**Review by gsanchezgavier — APPROVED** [review]

File: charts/super-agent-deployment/ci/test-values.yaml [hunk 31]
```diff
@@ -0,0 +1,5 @@
+global:
+  licenseKey: 1234567890abcdef1234567890abcdef12345678
+
+# The agent is going to test that is able to call OpAmp service so we set a valid server to connect to.
```
Comment (paologallinaharbur) [inline charts/super-agent-deployment/ci/test-values.yaml:4]: we could test it without opamp not having opamp defined

File: charts/super-agent-deployment/templates/_helpers.tpl (hunk 29 above)
Comment (kang-makes) [reply charts/super-agent-deployment/templates/_helpers.tpl:25]: Fair. For now, it is the best.

File: charts/super-agent-deployment/templates/_helpers.tpl (hunk 30 above)
Comment (kang-makes) [reply charts/super-agent-deployment/templates/_helpers.tpl]: You are completely right. Removed on [#032a7e5](https://github.com/newrelic/helm-charts/pull/1155/commits/032a7e5c51218d096b4842f6f469f89bb3566fd2)

File: charts/super-agent-deployment/templates/_helpers.tpl (hunk 29 above)
Comment (kang-makes) [reply charts/super-agent-deployment/templates/_helpers.tpl]: Done on [#032a7e5](https://github.com/newrelic/helm-charts/pull/1155/commits/032a7e5c51218d096b4842f6f469f89bb3566fd2)

**Review by paologallinaharbur — APPROVED** [review]
🎉
